ludwig/utils/defaults.py

```diff
@@ -130,7 +130,55 @@ def get_default_optimizer_params(optimizer_type):
         raise ValueError('Incorrect optimizer type: ' + optimizer_type)
 
 
+def _perform_sanity_checks(model_definition):
+    assert 'input_features' in model_definition, (
+        'Model definition does not define any input features'
+    )
+
+    assert 'output_features' in model_definition, (
+        'Model definition does not define any output features'
+    )
+
+    assert isinstance(model_definition['input_features'], list), (
+        'Ludwig expects input features in a list. Check your model '
+        'definition format'
+    )
+
+    assert isinstance(model_definition['output_features'], list), (
+        'Ludwig expects output features in a list. Check your model '
+        'definition format'
+    )
+
+    assert len(model_definition['input_features']) > 0, (
+        'Model definition needs to have at least one input feature'
+    )
+
+    assert len(model_definition['output_features']) > 0, (
+        'Model definition needs to have at least one output feature'
+    )
+
+    if 'training' in model_definition:
+        assert isinstance(model_definition['training'], dict), (
+            'There is an issue while reading the training section of the ' 
```

Comment:
Please specify here and in the next 2 messages that the value should be a dictionary